### ISSUE TITLE
Some ideas for enhancing the choice panel.

### ISSUE THREAD

**Issue by lodici** [magarena/magarena#706]
I am looking to update the style of the action buttons in the user prompt to match those used elsewhere. While looking at this I started playing around with the choice panels and have come up with two variations as shown below. The first image shows what is  currently implemented.

I think the two variations are better suited to variable and possibly lengthy list of choices than the current implementation, with the combo being the simplest and most flexible option but which does not look as good as the icons. The combo would also require two clicks - one to select and one to click the right arrow.

This would also be an opportunity to improve the text shown for each choice which at the moment is concise to say the least.

![screenshot00489](https://cloud.githubusercontent.com/assets/2443519/13191158/ab07d99c-d759-11e5-8230-17ef2dedd5ec.png)
![screenshot00488](https://cloud.githubusercontent.com/assets/2443519/13191214/17102e64-d75a-11e5-9de9-bb376f7ff3fb.jpg)
![screenshot00490](https://cloud.githubusercontent.com/assets/2443519/13191162/b2761f22-d759-11e5-806f-30d821a3ceed.png)

**Comment by ShawnieBoy** [comment]
Anything that can allow more descriptive information about what the ability is, what it does and what it costs would be excellent.

They should contain all the text as appears on the card/s instead of sometimes having to choose from 'Pump' 'Pump' and 'Pump'

**Comment by lodici** [comment]
> Anything that can allow more descriptive information about what the ability is, what it does and what it costs would be excellent.

I think this would apply to ``ModeChoicePanel`` which currently displays a number and ``PlayChoicePanel`` which currently displays a keyword as shown above. Currently, both receive limited model info that can be used by the UI.

``PlayChoicePanel`` gets its keyword from ``MagicPlayChoiceResult.getText()`` so if this can be updated to return a caption and optional description that would be helpful.

``ModeChoicePanel`` just receives a list of integers so again the model needs to be updated to provide more useful info that can be displayed in the UI.

**Comment by ShawnieBoy** [comment]
It's not so much of a problem with modal spells, as the spell is on the stack to refer to, and each effect is listed against it's corresponding number.

Things like the activated abilities of permanents are more confusing e.g. Goblin Sappers - both abilities have the same base effect, yet one costs more and the other also involves the Sappers being sacrificed as well.

The `MagicPlayChoiceResult.getText()` could be able to get the parsed text for scripted cards, or assemble it from combining the cost with the effect text that normally appears in the log.

**Comment by lodici** [comment]
> It's not so much of a problem with modal spells, as the spell is on the stack to refer to, and each effect is listed against it's corresponding number.

True. A "please see stack item" would be useful. Perhaps highlight the item as well.

> Things like the activated abilities of permanents are more confusing e.g. Goblin Sappers - both abilities have the same base effect, yet one costs more and the other also involves the Sappers being sacrificed as well.

Good example, this currently looks :-1: 
![screenshot00492](https://cloud.githubusercontent.com/assets/2443519/13196836/25380db8-d7d2-11e5-87a3-50ed23817849.png)
We should be able to display the actual mana icons. Also, is there any card which requires more than 4 mana since I would imagine based on the above screenshot, it will not fit.

**Comment by ShawnieBoy** [comment]
There's a series that needs 5 mana - turn into dragons, Dragonsoul Knight

**Comment by lodici** [comment]
Dragonsoul Knight has a single ability which does not trigger the choice UI.

**Comment by ShawnieBoy** [comment]
But if it gains another ability...

It is currently listed as pump, so doesn't display mana. The Goblin Sappers are groovy-fied so to be able to distinguish between the two abilities. If it was scripted it would list 'Unblockable' 'Unblockable'

**Comment by lodici** [comment]
> But if it gains another ability...

What is the best way to do that?

**Comment by ShawnieBoy** [comment]
Enchant it with Dragon Mantle - but that would show 'Becomes' 'Pump'

Forgot that Dragonsoul Knight is a 'becomes' ability

**Comment by ShawnieBoy** [comment]
Pushed TestAbilityChoice for an extreme example of multiple abilities with same/similar effects and different costs.

Target Filigree Sages with everything and try to determine which ability is which (The -are- in timestamp order, but that is a test of memory)

**Comment by ShawnieBoy** [comment]
I'd send a savegame so the timestamps are unknown (make it a bit trickier), but all the enchantments fall off (and it's no longer paired, or effected by Touch of Vitae)

**Comment by lodici** [comment]
Wow, that is quite horrible both from the choices offered and knowing what additional abilities the card has been given, although the log does help a lot in that case.
![screenshot00493](https://cloud.githubusercontent.com/assets/2443519/13197264/66a518fa-d7e0-11e5-8a97-ee61235c04e0.png)

**Comment by lodici** [comment]
> I'd send a savegame so the timestamps are unknown (make it a bit trickier), but all the enchantments fall off (and it's no longer paired, or effected by Touch of Vitae)

That is the major limitation of the save game mechanism as it stands.

**Comment by ShawnieBoy** [comment]
A creatureTypeChoice panel with the combobox would be very handy!
